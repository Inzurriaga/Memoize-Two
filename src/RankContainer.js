import React from "react"

export default function RankContainer
(props){
    let sortUsers = props.users.sort((usera, userb) => {
        return userb.completed.length - usera.completed.length
      })
    return(
        <div className="rank-section">
            <h2>player rank</h2>
        <ul>
            {
              sortUsers.map((user, index) => {
                return (
                    <li>
                        <p>#{index + 1}</p>
                        <p> {user.name} </p>
                        <p>{user.completed.length} completed</p>
                    </li>
                )
              })
            }
        </ul>
        </div>
    )
}