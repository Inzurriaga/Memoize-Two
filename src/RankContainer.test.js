import React from 'react';
import { shallow } from 'enzyme';
import RankContainer from "./RankContainer"


const Mockusers = [{"name":"duy","password":"duy","completed":[1,2,5]},{"name":"gabriel","password":"gabriel","completed":[1,2,5]},{"name":"isaac","password":"isaac","completed":[1,2]}]

describe("RankContainer", ()=> {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <RankContainer users={Mockusers}/>
        )
    })

    it("should pass test", () => {
        expect(wrapper).toMatchSnapshot()
    })
})