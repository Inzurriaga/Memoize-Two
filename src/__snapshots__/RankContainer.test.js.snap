// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankContainer should pass test 1`] = `
<div
  className="rank-section"
>
  <h2>
    player rank
  </h2>
  <ul>
    <li
      key="0"
    >
      <p>
        #
        1
      </p>
      <p>
         
        duy
         
      </p>
      <p>
        3
         completed
      </p>
    </li>
    <li
      key="1"
    >
      <p>
        #
        2
      </p>
      <p>
         
        gabriel
         
      </p>
      <p>
        3
         completed
      </p>
    </li>
    <li
      key="2"
    >
      <p>
        #
        3
      </p>
      <p>
         
        isaac
         
      </p>
      <p>
        2
         completed
      </p>
    </li>
  </ul>
</div>
`;
